VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/non-host-waiting-room/non-host-waiting-room.component.ts

```diff
@@ -0,0 +1,435 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, Subject } from 'rxjs';
+import { ConsultationService } from 'src/app/services/api/consultation.service';
+import { VideoWebService } from 'src/app/services/api/video-web.service';
+import { ConferenceStatus, ParticipantStatus, Role } from 'src/app/services/clients/api-client';
+import { ClockService } from 'src/app/services/clock.service';
+import { ErrorService } from 'src/app/services/error.service';
+import { EventsService } from 'src/app/services/events.service';
+import { Logger } from 'src/app/services/logging/logger-base';
+import { pageUrls } from 'src/app/shared/page-url.constants';
+import { DeviceTypeService } from '../../services/device-type.service';
+import { HearingRole } from '../models/hearing-role-model';
+import { NotificationSoundsService } from '../services/notification-sounds.service';
+import { NotificationToastrService } from '../services/notification-toastr.service';
+import { RoomClosingToastrService } from '../services/room-closing-toast.service';
+import { VideoCallService } from '../services/video-call.service';
+import { WaitingRoomBaseDirective } from '../waiting-room-shared/waiting-room-base.component';
+import { TranslateService } from '@ngx-translate/core';
+import { ConsultationInvitationService } from '../services/consultation-invitation.service';
+import { take, takeUntil } from 'rxjs/operators';
+import { UnloadDetectorService } from 'src/app/services/unload-detector.service';
+import { UserMediaService } from 'src/app/services/user-media.service';
+import { ParticipantMediaStatus } from 'src/app/shared/models/participant-media-status';
+import { Title } from '@angular/platform-browser';
+import { ModalTrapFocus } from '../../shared/modal/modal-trap-focus';
+import { HideComponentsService } from '../services/hide-components.service';
+import { FocusService } from 'src/app/services/focus.service';
+import { ConferenceState } from '../store/reducers/conference.reducer';
+import { Store } from '@ngrx/store';
+import { LaunchDarklyService } from 'src/app/services/launch-darkly.service';
+import { VHParticipant } from '../store/models/vh-conference';
+import { NonHostUserRole } from '../waiting-room-shared/models/non-host-user-role';
+
+@Component({
+    standalone: false,
+    selector: 'app-non-host-waiting-room',
+    templateUrl: './non-host-waiting-room.component.html',
+    styleUrls: ['../waiting-room-global-styles.scss', './non-host-waiting-room.component.scss']
+})
+export class NonHostWaitingRoomComponent extends WaitingRoomBaseDirective implements OnInit, OnDestroy {
+    @Input() userRole: NonHostUserRole;
+
+    currentTime: Date;
+    hearingStartingAnnounced: boolean;
+
+    isParticipantsPanelHidden = false;
+    hearingVenueIsScottish$: Observable<boolean>;
+
+    showWarning = false;
```

Comment:
rename to `showJoinHearingWarning` or a more verbose name so its clear what warning is to be shown